youtube-client/src/app/core/components/header/login/login.component.ts

```diff
@@ -1,10 +1,32 @@
-import { Component, Input } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import LoginService from 'src/app/auth/services/login.service';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss'],
 })
-export default class LoginComponent {
-  @Input() username: string | null;
+export default class LoginComponent implements OnInit {
+  username: string;
+
+  showButtonLogout: boolean;
+
+  constructor(private loginService: LoginService) {}
+
+  ngOnInit(): void {
+    this.loginService.loginName$.subscribe((value) => {
+      this.username = value;
+    });
+    this.loginService.isLogged$.subscribe((value) => {
+      this.showButtonLogout = !value;
+    });
```

Comment:
If these two properties are only used in the template, you could utilize the `| async` pipe to avoid subscribing to these observables